charts/camunda-platform-8.8/test/unit/camunda/no_secondary_storage_test.go

```diff
@@ -0,0 +1,74 @@
+// Copyright 2022 Camunda Services GmbH
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package camunda
+
+import (
+	"camunda-platform/test/unit/testhelpers"
+	"path/filepath"
+	"strings"
+	"testing"
+
+	"github.com/gruntwork-io/terratest/modules/random"
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+)
+
+type NoSecondaryStorageTemplateTest struct {
+	suite.Suite
+	chartPath string
+	release   string
+	namespace string
+	templates []string
+}
+
+func TestNoSecondaryStorageTemplate(t *testing.T) {
+	t.Parallel()
+
+	chartPath, err := filepath.Abs("../../../")
+	require.NoError(t, err)
+
+	suite.Run(t, &NoSecondaryStorageTemplateTest{
+		chartPath: chartPath,
+		release:   "camunda-platform-test",
+		namespace: "camunda-platform-" + strings.ToLower(random.UniqueId()),
+	})
+}
+
+func (s *NoSecondaryStorageTemplateTest) TestNoSecondaryStorageGlobalValue() {
+	testCases := []testhelpers.TestCase{
+		{
+			Name:                 "TestGlobalNoSecondaryStorageTogglesAllExpectedValues",
+			HelmOptionsExtraArgs: map[string][]string{"install": {"--debug"}},
+			Values: map[string]string{
+				"global.noSecondaryStorage": "true",
+			},
+			Verifier: func(t *testing.T, output string, err error) {
+				require.NoError(t, err)
+				// Database type should be none
+				require.Contains(t, output, "database:\n        type: none")
+				// Agentic AI and inbouond connectors should be disabled
+				require.Contains(t, output, "webhook:\n          enabled: false")
+				require.Contains(t, output, "polling:\n          enabled: false")
+				require.Contains(t, output, "agenticai:\n          enabled: false")
+				// Optimize should not be rendered
```

Comment:
It should be `templates/optimize` and `templates/connectors` as the apps are part of the main chart, not the sub-charts.

Also, the Optimize clue will not fail, as [we disable Optimize by default](https://github.com/camunda/camunda-platform-helm/blob/d750ed1c51a7d0f3d9aab904af8f7266cf9c7357/charts/camunda-platform-8.8/values.yaml#L2639), but if the user enabled it, it will be rendered.

The `global.noSecondaryStorage` check should be added to all component resources where the component is enabled, like:

https://github.com/camunda/camunda-platform-helm/blob/d750ed1c51a7d0f3d9aab904af8f7266cf9c7357/charts/camunda-platform-8.8/templates/optimize/deployment.yaml#L1